src/JobPanel/backend/async_repeater.py

```diff
@@ -318,6 +323,63 @@ def handle_close(self):
         return
 
 
+class StarterServer(asyncore.dispatcher):
+    def __init__(self, ar):#,  repeater_address, port, clients):
+        """
+        host - get automatically
+        :param port - port ( same as in socket module)
+        """
+        asyncore.dispatcher.__init__(self)
+        #self.server_dispatcher = ServerDispatcher(repeater_address, port, clients)
+
+        self.ar = ar
+
+        self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.bind(("", 0))
+        self.address=self.socket.getsockname()
+        #self.answers=[]
+        self.listen(5)
+
+
+    # def get_dispatcher(self):
+    #     return self.server_dispatcher
+
+    # def handle_accept(self):
+    #     # Called when a client connects to our socket
+    #     client_info = self.accept()
+    #     logging.info("Incomming connection accepted.\n")
+    #     self.server_dispatcher.accept(client_info[0])
+    #     return
+
+    def handle_accepted(self, sock, addr):
+        StarterServerDispatcher(sock, self.ar)
+        #print(sock.getsockname())
+
+    # def handle_close(self):
+    #     self.close()
+
+
+class StarterServerDispatcher(asyncore.dispatcher):
+    def __init__(self, sock, ar):
+        super().__init__(sock)
+        self.ar = ar
+
+    def handle_read(self):
+        data = self.recv(1024)
+        #print(data)
+        if data:
+            s = data.decode().split("\n", maxsplit=2)
+            if len(s) >= 2:
+                child_id = int(s[0])
+                port = int(s[1])
+                print((child_id, port))
+                # je to paralelne bezpecne??????
+                #print(self.ar.clients)
+                if self.ar.clients[child_id] is None:
```

Comment:
Client is already created in add_child so this is never true. The _ClientDispatcher have three states:

1. Waiting for address (considered None here), should have self._remote_address=[], get_remote_address() return None

2. Waiting for connection call, have valid self._remote_address, but not connected yet (can not send or recieve message) 

3. Fully connected.